src/main/java/frc/robot/subsystems/base/common/SensorBase.java

```diff
@@ -0,0 +1,43 @@
+package frc.robot.subsystems.base.common;
+
+/**
+ * Add methods that indicated that the subsystem that sensors are major part of the subsystem.
+ * <p>
+ *
+ * @author Barel
+ * @version 1.0
+ * @since 2020-off
+ */
+public interface SensorBase {
```

Comment:
```suggestion
public interface SensorBased {
```
A subsystem is _based on_ a sensor, it is not _a_ sensor.